src/commands/new.rs

```diff
@@ -352,11 +353,25 @@ fn path_safe(text: &str) -> PathBuf {
 }
 
 fn validate_name(name: &str) -> Result<String, String> {
-    if NAME.is_match(name) {
-        Ok(name.to_owned())
-    } else {
-        Err("Name must start with a letter".to_owned())
+    let splits = name.split(|c| !char::is_alphanumeric(c));
```

Comment:
Can the separator legally be any non-alphanumeric character? Is "foo😴a" a legal name?

Reply:
Well, it works, so I guess it's de facto legal!  The name as written is only used for the app name, which Spin doesn't care about.  Component names get kebab-cased during processing, which turns that into `foo-a` which is a legal kebab-id.  And the generated directory gets snake-cased to `foo_a` which is a legal directory name.